manticore/core/manticore.py

```diff
@@ -278,6 +283,69 @@ def _join_workers(self):
             while len(self._workers) > 0:
                 self._workers.pop().join()
 
+    #
+    # Symbol resolution
+    #
+
+    def _addr_by_symline(dwarfinfo, symbols, line):
+        '''
+        Return address for a symbol and line it is on through DWARF
+        debug info
+        '''
+        for symbol in symbols:
+            for CU in dwarfinfo.iter_CUs():
+                lineprog = dwarfinfo.line_program_for_CU(CU)
+                prevstate = None
+
+                for entry in lineprog.get_entries():
+                    # check for new state assignment
+                    if entry.state is None or entry.state.end_sequence:
+                        continue
+                    # check between address range between prev and current state containing address
+                    if prevstate and prevstate.address <= symbol['st_value'] < entry.state.address:
+                        state_line = prevstate.line
+                        if int(state_line) is line:
+                            return symbol['st_value']
+                    prevstate = entry.state
+        
+        return None, None
+
+
+    def resolve(self, symbol, line=None):
+        '''
+        A helper method used to resolve a symbol name into a memory address when
+        injecting hooks for analysis. 
+
+        :param symbol: function name to be resolved
+        :type symbol: string
+
+        :param line: if more functions present, optional line number can be included
+        :type line: int or None
+        '''
+        with open(self._path, 'rb') as f:
+
+            elffile = ELFFile(f)
+            if elffile is None:
+                return NotImplementedError("Symbols aren't supported")
```

Comment:
Can this ever happen?

I believe `ELFFile` is a class and since `__init__` can't really return `None` this will probably raise an exception instead of leaving us with `None`. As in:

* `ELFFile.__init__` calls https://github.com/eliben/pyelftools/blob/master/elftools/elf/elffile.py#L65-L82
* `ELFFile._identify_file` makes some assertions https://github.com/eliben/pyelftools/blob/master/elftools/elf/elffile.py#L409-L417

Reply:
Thanks for the info! `ELFFile._identify_file` calls `elf_assert`, which raises `ELFError`, so the error-checking on part isn't necessary.